src/createAnimatedComponent/NativeEventsManager.ts

```diff
@@ -0,0 +1,144 @@
+    if (this._eventViewTag !== computedEventTag) {
+      // Remove all bindings from previous props that ran on the old viewTag
+      executeForEachEventHandler(prevProps, (_key, handler) => {
+        handler.unregisterFromEvents(this._eventViewTag);
+      });
+      // We don't need to unregister from current (new) props, because their events weren't registered yet
+      // Replace the view tag
+      this._eventViewTag = computedEventTag;
+      // Attach the events with a new viewTag
+      this.attachNativeEvents();
+      return;
+    }
+
+    executeForEachEventHandler(prevProps, (key, prevHandler) => {
+      const newProp = this._managedComponent.props[key];
+      if (!newProp) {
+        // Prop got deleted
+        prevHandler.unregisterFromEvents(this._eventViewTag);
+      } else if (
+        isWorkletEventHandler(newProp) &&
+        newProp.workletEventHandler !== prevHandler
+      ) {
+        // Prop got changed
+        prevHandler.unregisterFromEvents(this._eventViewTag);
+        newProp.workletEventHandler.registerForEvents(this._eventViewTag);
+      }
+    });
+
+    executeForEachEventHandler(this._managedComponent.props, (key, handler) => {
+      if (!prevProps[key]) {
+        // Prop got added
+        handler.registerForEvents(this._eventViewTag);
+      }
+    });
+  }
+}
+
+type ManagedAnimatedComponent = React.Component<
+  AnimatedComponentProps<InitialComponentProps>
+> &
+  IAnimatedComponentInternal;
+
+type ComponentOptions = {
+  setNativeProps: (
+    ref: AnimatedComponentRef,
+    props: InitialComponentProps
+  ) => void;
+};
+
+type WorkletEventHandlerHolder = {
+  workletEventHandler: InstanceType<typeof WorkletEventHandler>;
+};
+
+function isWorkletEventHandler(
+  prop: unknown
+): prop is WorkletEventHandlerHolder {
+  return (
+    has('workletEventHandler', prop) &&
+    prop.workletEventHandler instanceof WorkletEventHandler
+  );
+}
+
+function executeForEachEventHandler(
+  props: AnimatedComponentProps<InitialComponentProps>,
+  callback: (
+    key: string,
+    handler: InstanceType<typeof WorkletEventHandler>
+  ) => void
+) {
+  for (const key in props) {
+    const prop = props[key];
+    if (isWorkletEventHandler(prop)) {
+      callback(key, prop.workletEventHandler);
+    }
+  }
+}
+
+function getEventViewTag(
+  component: ManagedAnimatedComponent,
+  options?: ComponentOptions
+) {
+  // Get the tag for registering events - since the event emitting view can be nested inside the main component
+  const componentAnimatedRef = component._component as AnimatedComponentRef;
+  let newTag: number;
+  if (componentAnimatedRef.getScrollableNode) {
+    const scrollableNode = componentAnimatedRef.getScrollableNode();
+    newTag = findNodeHandle(scrollableNode) ?? -1;
+  } else {
+    newTag =
+      findNodeHandle(
+        options?.setNativeProps ? component : componentAnimatedRef
+      ) ?? -1;
+  }
+  return newTag;
+}
```

Comment:
Same here, instead of passing the `component` we could make it a private method.